analyzers/src/SonarAnalyzer.CSharp/Rules/FieldsShouldBeEncapsulatedInProperties.cs

```diff
@@ -37,6 +37,10 @@ public sealed class FieldsShouldBeEncapsulatedInProperties : SonarDiagnosticAnal
         SyntaxKind.ConstKeyword
     };
 
+    private static readonly ImmutableArray<KnownType> IgnoredTypes = ImmutableArray.Create(
+        KnownType.UnityEngine_MonoBehaviour,
+        KnownType.UnityEngine_ScriptableObject);
```

Comment:
My understanding from the docs is that you should include `UnityEngine.Object`.
Which, if I'm not wrong, is parent to `UnityEngine_ScriptableObject` so you can replace one with the other.

Reply:
Adding `UnityEngine.Object` is not enough, the class should also have the `Serializable` attribute.
I can add it so it also supports custom serializable unity class/object.
I will rework a bit the logic then.

Reply:
After a closer look, the required attribute for custom objects is `System.Serializable`, and this is already an exception to the rule.